lib/CL/devices/bufalloc.c

```diff
@@ -104,7 +104,7 @@ chunk_slack (chunk_info_t* chunk, size_t size, size_t* last_chunk_size)
   size_t end_addr = aligned_start_addr + size;
   if(last_chunk_size)
     *last_chunk_size = end_chunk - end_addr;
-  return end_chunk > end_addr;
```

Comment:
Why this change? What is the broken logic in the original?

Reply:
In a situation where 1) you have X bytes of (aligned) free device memory in `region->last_chunk` 2) you call  `chunk_slack(region->last_chunk, X, NULL);` the original code returns 0 (false). IOW at most X-1 free bytes are usable.

Reply:
OK. Hard to remember anymore.